### PR TITLE
Curation task email parameter changes

### PR DESCRIPTION
**Pull request by jonas-atmire**
## References
* Fixes [PR comment issue](https://github.com/DSpace/DSpace/pull/2820#pullrequestreview-459310333)
* Related to [similar issue](https://github.com/DSpace/DSpace/issues/2822)/[PR](https://github.com/DSpace/DSpace/pull/2915)
## Description
This PR will remove the "email" option from the REST curate script. The email parameter is still available (& mandatory) for the CLI implementation of this script. The REST implementation of the curate script will use the currently logged in user to execute the script.

## Instructions for Reviewers

Retrieve the configuration of the REST curate script, the email option should not be available here. In the CLI the option will be present & will be mandatory.

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome). If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_

- [c] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & integration tests). Exceptions may be made if previously agreed upon.
- [x] My PR passes Checkstyle validation based on the [Code Style Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Style+Guide).
- [x] My PR includes Javadoc for _all new (or modified) public methods and classes_. It also includes Javadoc for large or complex private methods.
- [x] My PR passes all tests and includes new/updated Unit or Integration Tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [x] If my PR includes new, third-party dependencies (in any `pom.xml`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
- [x] If my PR modifies the REST API, I've linked to the REST Contract page (or open PR) related to this change.

### CODE REVIEW

**Comment by tdonohue** [comment]
@jonas-atmire : It appears this PR was accidentally overlooked/lost as it's not on our Project Board.  I've just now added it, so we can get it assigned for review.  However, it looks like it needs _merge conflicts resolved_ before we can review/test it.

**Review by tdonohue — APPROVED** [review]
👍 Code looks good. I've also tested it both from the REST API and from commandline.  The `-e` is not allowed via REST, but is required via CLI. So, this is working as expected.  Thanks @jonas-atmire !

That said, I noticed that with this PR in place, the *Angular UI* Curation Tasks feature now fails.  It looks like that feature always send the `-e` parameter, which obviously now fails in the REST API.  So, merging this PR will break the http://localhost:4000/admin/curation-tasks interface.  (That said, the `/processes` interface in the UI works great still..as it gets the params dynamically, it doesn't allow you to add the `-e` to the curate process)

**In summary, this backend PR looks great. However, we need a corresponding Angular UI fix (it's likely a very small fix).** If you are unable to provide that Angular PR, maybe we can ask @artlowel to assign it to someone else?

**Comment by mwoodiupui** [comment]
I wonder whether not-allowed-in-REST options should result in failure, or just be ignored (perhaps with a warning).

Regardless, the Curation Tasks feature should be fixed do the Right Thing.

**Comment by tdonohue** [comment]
@mwoodiupui : While I understand the desire to not throw an error, I think it's appropriate here...otherwise, if someone is running these tasks in REST API via `curl` or `wget` or similar, they may expect one behavior (their task will run under the Eperson they pass along) when it fact it kicks off under a different EPerson (the one which they are authenticated as).  If we only threw a warning, there'd be no way for them to "undo" or "correct" that request...it'd just ignore the `-e` param & start the process.  Additionally, another major issue here is that I'm not sure how easy it'd be in our REST API to return a successful 200 response code, along with a warning message...that's not a normal RESTful response, where 200 = OK.

In any case, the necessary fix here is really really minor (may literally be a one line fix).  We just need a tiny fix to the Curation Task UI code to no longer pass in the `-e` param to the backend.

**Comment by jonas-atmire** [comment]
@tdonohue 
Sorry, I had missed your initial mention about the backend being OK, but it needing a fix for the UI.
I created the [Angular related PR](https://github.com/DSpace/dspace-angular/pull/855) quickly, since as you mentioned, this was quite a small change.

**Comment by tdonohue** [comment]
@jonas-atmire : Thanks! I've retested this with https://github.com/DSpace/dspace-angular/pull/855 and everything now works!

Merging this & corresponding Angular PR as this was previously flagged for `1 Approval`
